azkaban-exec-server/src/main/java/azkaban/execapp/FlowPreparer.java

```diff
@@ -271,8 +275,13 @@ void setupProject(final ProjectVersion pv)
       if (this.projectDirCleaner != null) {
         this.projectDirCleaner.deleteProjectDirsIfNecessary(pv.getDirSizeInBytes());
       }
-      Files.move(tempDir.toPath(), pv.getInstalledDir().toPath(), StandardCopyOption.ATOMIC_MOVE);
+      try {
+        Files.move(tempDir.toPath(), pv.getInstalledDir().toPath(), StandardCopyOption.ATOMIC_MOVE);
```

Comment:
Is it possible that the other thread, which downloaded the same project directory, could delete the directory before this thread has a chance to hard link?